### PR TITLE
chore: Switch to typedoc@beta for documentation generation

### PR DESCRIPTION
**Pull request by Gerrit0**
The external module name plugin is far more powerful than is actually needed here... and the custom version included here was overkill, so I replaced it with a minimal version that works properly.

There is one weird artifact, I have no idea why TypeDoc isn't removing `@preferred` from comments... I don't remember changing that code at all.

### CODE REVIEW

**Comment by Gerrit0** [comment]
CI failing appears to be unrelated, broken on master too.

I dug into the `@preferred` tag showing up, and it appears that this behavior hasn't changed in a while... at least 7 months, so I'm not sure why it shows up here but not on the older build. In either case, it shouldn't be needed here.

File: typedoc-plugin-external-module-name/index.js [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+const { Converter } = require('typedoc/dist/lib/converter');
+
+/** @param {import("typedoc/dist/lib/utils/plugins").PluginHost} host */
+exports.load = function (host) {
+  host.application.converter.on(Converter.EVENT_RESOLVE_BEGIN, context => {
+    /** @type {import("typedoc").ProjectReflection} */
+    const project = context.project;
+
+    for (const mod of project.children.filter(child => child.kind === ReflectionKind.Module)) {
+      const tag = mod.comment?.getTag('module');
```
Comment (MasterKale) [inline typedoc-plugin-external-module-name/index.js:14]: I figured out why Netlify build is failing - this JS gets executed in Node v10.22.1 which of course doesn't support optional chaining:

```
2:21:15 PM: Error: The plugin /opt/build/repo/typedoc-plugin-external-module-name could not be loaded.
2:21:15 PM: /opt/build/repo/typedoc-plugin-external-module-name/index.js:14
2:21:15 PM:       const tag = mod.comment?.getTag('module');
2:21:15 PM:                               ^
2:21:15 PM: SyntaxError: Unexpected token .
2:21:15 PM:     at Module._compile (internal/modules/cjs/loader.js:723:23)
2:21:15 PM:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
2:21:15 PM:     at Module.load (internal/modules/cjs/loader.js:653:32)
2:21:15 PM:     at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
2:21:15 PM:     at Function.Module._load (internal/modules/cjs/loader.js:585:3)
2:21:15 PM:     at Module.require (internal/modules/cjs/loader.js:692:17)
2:21:15 PM:     at require (internal/modules/cjs/helpers.js:25:18)
2:21:15 PM:     at PluginHost.load (/opt/build/repo/node_modules/typedoc/dist/lib/utils/plugins.js:34:34)
2:21:15 PM:     at CliApplication.bootstrap (/opt/build/repo/node_modules/typedoc/dist/lib/application.js:84:22)
2:21:15 PM:     at CliApplication.bootstrap (/opt/build/repo/node_modules/typedoc/dist/lib/cli.js:33:30)
2:21:15 PM: npm ERR! code ELIFECYCLE
2:21:15 PM: npm ERR! errno 1
2:21:15 PM: npm ERR! simplewebauthn-monorepo@ docs: `npm run bootstrap && typedoc --tsconfig tsconfigdoc.json`
2:21:15 PM: npm ERR! Exit status 1
2:21:15 PM: npm ERR!
2:21:15 PM: npm ERR! Failed at the simplewebauthn-monorepo@ docs script.
2:21:15 PM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
2:21:15 PM: npm ERR! A complete log of this run can be found in:
2:21:15 PM: npm ERR!     /opt/buildhome/.npm/_logs/2020-10-25T21_21_15_629Z-debug.log
```

**Comment by MasterKale** [comment]
@Gerrit0 Are you able to replace that optional chain with more vanilla Node-compatible code? That should let the docs build so I can compare them to what we have now.

**Comment by Gerrit0** [comment]
Generally don't have much time during the week - this might work. 

No idea why I thought master was failing... I must have been looking at a different repo.

**Comment by MasterKale** [comment]
@Gerrit0: I just realized you're one of _the_ TypeDoc devs, thank you for stopping by and contributing. Now that I can preview the changes this is definitely looking like an improvement!

I'm going back and forth between the current docs and the preview of these docs and am noticing a few instances of types in Typescript's **lib.dom.d.ts** not being linked. It seems to be when I reference one in an `Omit<>` ([PublicKeyCredentialRequestOptionsJSON](https://deploy-preview-66--simplewebauthn.netlify.app/interfaces/_simplewebauthn_typescript_types_.publickeycredentialrequestoptionsjson.html)):

Screen Shot 2020-10-28 at 7 20 16 AM

(PublicKeyCredentialRequestOptions is declared in TypeScript's DOM lib [here](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L1128))

I'll create an Issue in the actual TypeDoc repo for this, though 😄

**Pull request merged**